### PR TITLE
Deal with space after +/-

### PR DESCRIPTION
**Pull request by billdueber**
...when doing solr escaping.

Note this does _not_ yet test if actually works with solr; coming next.

### CODE REVIEW

**Comment by antmoth** [comment]
All the backslashes making my head hurt, but if the tests pass this seems like a reasonable start?

File: lib/mlibrary_search_parser/search_handler.rb [hunk 1]
```diff
@@ -2,13 +2,51 @@
 require 'dotenv/load'
 
 module MLibrarySearchParser
-  class UnevenParensError < RuntimeError; end
+  class ParseError
+    DETIALS = "ParseError"
```
Comment (antmoth) [inline lib/mlibrary_search_parser/search_handler.rb]: probably needs to be DETAILS

File: lib/mlibrary_search_parser/transform/solr/local_params.rb [hunk 2]
```diff
@@ -13,6 +13,7 @@ def transform!
           else
             super
             set_param("q", "_query_:#{query}")
+            set_param("clean_string", original_search_tree.clean_string)
```
Comment (antmoth) [inline lib/mlibrary_search_parser/transform/solr/local_params.rb]: I think we don't need this any more

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -45,6 +45,8 @@ def not_node(node_or_string)
 @config_file = './spec/data/00-catalog.yml'
 @config = YAML.load(ERB.new(File.read(@config_file)).result)
 TEST_HANDLER =  MLibrarySearchParser::SearchHandler.new(@config)
+SEARCH_FACTORY =  MLibrarySearchParser::Search.search_builder(@config)
```
Comment (antmoth) [inline spec/spec_helper.rb]: is this being used?

**Review by antmoth — COMMENTED** [review]
I flagged a couple small things that aren't actually related to the lucene escaping but overall this looks fine

File: lib/mlibrary_search_parser/transform/solr/local_params.rb (hunk 2 above)
Comment (billdueber) [reply lib/mlibrary_search_parser/transform/solr/local_params.rb]: Changing it to a method on solr transforms so you can do `localparams_obj.clean_string`, parallel with what we do with everything else.

**Pull request merged**